Modules/_io/_iomodule.c

```diff
@@ -532,7 +532,7 @@ PyNumber_AsOff_t(PyObject *item, PyObject *err)
 {
     Py_off_t result;
     PyObject *runerr;
-    PyObject *value = PyNumber_Index(item);
+    PyObject *value = _PyNumber_Index(item);
```

Comment:
I'm missing something here: from my reading of the code, `PyLong_AsOff_t` is either `PyLong_AsLong` or `PyLong_AsLongLong`, and both of those already call `_PyNumber_Index`, so it looks as though we don't need this call.

Ah, I think I found what I'm missing: `PyLong_AsOff_t` could be `PyLong_AsSsize_t`, which _doesn't_ call `_PyNumber_Index`. Is that right?

Reply:
Yes, it is right.

It a separate question whether `PyLong_AsSsize_t` and others should call `__index__` implicitly. It is not simple question, because such change may introduce bugs by making presumably atomic code non-atomic.

Reply:
> It a separate question whether `PyLong_AsSsize_t` and others should call `__index__` implicitly.

Agreed. It seems worth looking into, but not in this PR.